ts/a11y/explorer/Highlighter.ts

```diff
@@ -0,0 +1,485 @@
+    return Array.from(
+      node.getElementsByClassName(this.mactionName)
+    ) as HTMLElement[];
+  }
+
+  /**
+   * @override
+   */
+  public isMactionNode(node: Element): boolean {
+    const className = node.className || node.getAttribute('class');
+    return className ? !!className.match(new RegExp(this.mactionName)) : false;
+  }
+
+  /**
+   * Check if a node is already highlighted.
+   *
+   * @param {HTMLElement} node The node.
+   * @returns {boolean} True if already highlighted.
+   */
+  public isHighlighted(node: HTMLElement): boolean {
+    return node.hasAttribute(this.ATTR);
+  }
+
+  /**
+   * Sets the indicator attribute that node is already highlighted.
+   *
+   * @param {HTMLElement} node The node.
+   */
+  public setHighlighted(node: HTMLElement) {
+    node.setAttribute(this.ATTR, 'true');
+  }
+
+  /**
+   * Removes the indicator attribute that node is already highlighted.
+   *
+   * @param {HTMLElement} node The node.
+   */
+  public unsetHighlighted(node: HTMLElement) {
+    node.removeAttribute(this.ATTR);
+  }
+}
+
+class SvgHighlighter extends AbstractHighlighter {
+  /**
+   * @override
+   */
+  constructor() {
+    super();
+    this.mactionName = 'maction';
+  }
+
+  /**
+   * @override
+   */
+  public highlightNode(node: HTMLElement) {
+    let info: Highlight;
+    if (this.isHighlighted(node)) {
+      info = {
+        node: node,
+        background: this.background,
+        foreground: this.foreground,
+      };
+      return info;
+    }
+    if (node.tagName === 'svg' || node.tagName === 'MJX-CONTAINER') {
+      info = {
+        node: node,
+        background: node.style.backgroundColor,
+        foreground: node.style.color,
+      };
+      node.style.backgroundColor = this.background;
+      node.style.color = this.foreground;
+      return info;
+    }
+    // This is a hack for v4.
+    // TODO: v4 Change
+    // const rect = (document ?? DomUtil).createElementNS(
+    const rect = document.createElementNS('http://www.w3.org/2000/svg', 'rect');
+    rect.setAttribute(
+      'sre-highlighter-added', // Mark highlighting rect.
+      'true'
+    );
+    const padding = 40;
+    const bbox: SVGRect = (node as any as SVGGraphicsElement).getBBox();
+    rect.setAttribute('x', (bbox.x - padding).toString());
+    rect.setAttribute('y', (bbox.y - padding).toString());
+    rect.setAttribute('width', (bbox.width + 2 * padding).toString());
+    rect.setAttribute('height', (bbox.height + 2 * padding).toString());
+    const transform = node.getAttribute('transform');
+    if (transform) {
+      rect.setAttribute('transform', transform);
+    }
+    rect.setAttribute('fill', this.background);
+    node.setAttribute(this.ATTR, 'true');
+    node.parentNode.insertBefore(rect, node);
+    info = { node: node, foreground: node.getAttribute('fill') };
+    if (node.nodeName !== 'rect') {
+      // We currently do not change foreground of collapsed nodes.
```

Comment:
I'm not sure I understand how the condition in the previous line relates to this.  And in my testing, the foreground color of collapsed nodes does get changed.

Reply:
> But there is still the timing issue wrt. highlighting `maction` backgrounds

That is resolved in one of my PRs.